runelite-client/src/main/java/net/runelite/client/plugins/microbot/zerozero/varrockcleaner/VarrockCleanerScript.java

```diff
@@ -0,0 +1,105 @@
+package net.runelite.client.plugins.microbot.zerozero.varrockcleaner;
+
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.plugins.microbot.Microbot;
+import net.runelite.client.plugins.microbot.Script;
+import net.runelite.client.plugins.microbot.util.equipment.Rs2Equipment;
+import net.runelite.client.plugins.microbot.util.gameobject.Rs2GameObject;
+import net.runelite.client.plugins.microbot.util.keyboard.Rs2Keyboard;
+import net.runelite.client.plugins.microbot.util.inventory.Rs2Inventory;
+import net.runelite.client.plugins.microbot.util.player.Rs2Player;
+import net.runelite.client.plugins.microbot.util.widget.Rs2Widget;
+
+import java.util.concurrent.TimeUnit;
+
+public class VarrockCleanerScript extends Script {
+
+    private enum State {
+        TAKE_UNCLEANED,
+        CLEAN_FIND,
+        STORAGE_CRATE,
+        DROP_ITEMS
+    }
+
+    private State currentState = State.TAKE_UNCLEANED;
+
+    public boolean run(VarrockCleanerConfig config) {
+        stop(); // Ensures any previous instance of the task is cancelled before starting
+        currentState = State.TAKE_UNCLEANED; // Explicitly reset to initial state
+
+        mainScheduledFuture = scheduledExecutorService.scheduleWithFixedDelay(() -> {
+            try {
+                if (!Microbot.isLoggedIn() || !super.run()) return;
+
+                switch (currentState) {
+                    case TAKE_UNCLEANED:
+                        takeUncleanedFinds();
+                        break;
+                    case CLEAN_FIND:
+                        cleanFinds();
+                        break;
+                    case STORAGE_CRATE:
+                        storeFindsInCrate();
+                        break;
+                    case DROP_ITEMS:
+                        dropUnwantedItems();
+                        break;
+                }
+
+            } catch (Exception ex) {
+                Microbot.log(ex.getMessage());
+            }
+        }, 0, 100, TimeUnit.MILLISECONDS);
+
+        return true;
+    }
+
+    private void takeUncleanedFinds() {
+        if (Rs2Inventory.isFull()) {
+            currentState = State.CLEAN_FIND;
+            return;
+        }
+        if (Rs2GameObject.interact(24557, "Take")) {
+            sleepUntil(() -> !Rs2Inventory.isFull(), 5000);
+        }
+        if (Rs2Inventory.isFull()) {
+            currentState = State.CLEAN_FIND;
+        }
+    }
+
+    private void cleanFinds() {
+        if (Rs2Inventory.contains("Uncleaned find") && Rs2GameObject.interact(24556, "Clean")) {
```

Comment:
Try to use ObjectID class as much as possible.

24556 can be written as: ObjectID.SPECIMEN_TABLE_24556

Reply:
All done